sor/src/main/java/com/bazaarvoice/emodb/sor/db/astyanax/CqlDataWriterDAO.java

```diff
@@ -260,7 +268,8 @@ public void onSuccess(@Nullable ResultSet result) {
 
             @Override
             public void onFailure(Throwable t) {
-                Throwables.propagate(t);
+                semaphore.release();
+                error.set(t);
```

Comment:
Don't need to change this, just an FYI.  If we're only interested in capturing the _first_ exception then you could do:

`error.compareAndSet(null, t);`

As it is now it captures the last error if there is more than one.  Not sure it matters in this case, but something to keep in mind in the future.